src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,294 @@
+package duke;
+
+import duke.data.exception.DukeException;
+import duke.data.task.Deadline;
+import duke.data.task.Event;
+import duke.data.task.Task;
+import duke.data.task.ToDo;
+
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+class Duke {
```

Comment:
Generally no problem with the code 👍 
However try not to rely on deep nesting and long methods as they affect readability